### PR TITLE
DMVP-1232: add tests to the module

### PR DESCRIPTION
**Pull request by aramkarapetian**
https://tutorbot.atlassian.net/browse/DMVP-1232

### CODE REVIEW

File: modules/dashboard/modules/widgets/container/network/base.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module "base" {
+  source = "../../base"
+
+  coordinates = var.coordinates
+
+  name = "Container Memory / ${var.container}"
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/base.tf]: it seems the name is wrong, here should be "Container Network"

File: modules/dashboard/modules/widgets/container/network/variables.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
+variable "container" {
+  type = string
+}
+
+variable "cluster" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/variables.tf:5]: unused variable

File: modules/dashboard/modules/widgets/container/network/variables.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  type = string
+}
+
+variable "namespace" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/variables.tf:9]: unused variable

File: modules/dashboard/modules/widgets/container/network/variables.tf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "default"
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/network/variables.tf [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/network/variables.tf [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/network/variables.tf:25]: unused variable

File: modules/dashboard/modules/widgets/container/cpu/variables.tf [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  type = string
+}
+
+variable "namespace" {
+  type    = string
+  default = "default"
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/cpu/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/cpu/variables.tf [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/cpu/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/cpu/variables.tf [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/cpu/variables.tf:25]: unused variable

File: modules/dashboard/modules/widgets/container/memory/variables.tf [hunk 10]
```diff
@@ -0,0 +1,39 @@
+variable "container" {
+  type = string
+}
+
+variable "cluster" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/memory/variables.tf:5]: unused variable

File: modules/dashboard/modules/widgets/container/memory/variables.tf [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  type = string
+}
+
+variable "namespace" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/memory/variables.tf:9]: unused variable

File: modules/dashboard/modules/widgets/container/memory/variables.tf [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "default"
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/memory/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/memory/variables.tf [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/memory/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/memory/variables.tf [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/memory/variables.tf:25]: unused variable

File: modules/dashboard/modules/widgets/container/restarts/variables.tf [hunk 15]
```diff
@@ -0,0 +1,39 @@
+variable "container" {
+  type = string
+}
+
+variable "cluster" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/restarts/variables.tf:5]: unused variable

File: modules/dashboard/modules/widgets/container/restarts/variables.tf [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+  type = string
+}
+
+variable "namespace" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/restarts/variables.tf:9]: unused variable

File: modules/dashboard/modules/widgets/container/restarts/variables.tf [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "default"
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/restarts/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/restarts/variables.tf [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/restarts/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/container/restarts/variables.tf [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/container/restarts/variables.tf:25]: unused variable

File: modules/dashboard/tests/widgets-balancer/1-example.tf [hunk 20]
```diff
@@ -0,0 +1,47 @@
+# data "aws_vpcs" "vpcs" {}
```
Comment (mrdntgrn) [inline modules/dashboard/tests/widgets-balancer/1-example.tf]: commented code

File: modules/dashboard/tests/dashboard-layout/test.yaml [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+    namespace : "default"
+    clustername : "dasmeta-test-new3"
+    width : 6
+    height : 5
+rows:
+    -
+      - name: "Text"
+        type : "text"
+        content : "Nginx"
+    # -
```
Comment (mrdntgrn) [inline modules/dashboard/tests/dashboard-layout/test.yaml]: commented code

File: modules/dashboard/tests/widgets-balancer/1-example.tf [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+module "dashboard-with-balancer-metrics" {
+  source = "../../"
+  name   = "dashboard-with-balancer-metrics"
+
+  defaults = {
+    period : 300
+  }
+
+  rows = [
+    # [
```
Comment (mrdntgrn) [inline modules/dashboard/tests/widgets-balancer/1-example.tf:22]: commented code

File: modules/dashboard/modules/widgets/balancer/2xx/balancer.tf [hunk 23]
```diff
@@ -0,0 +1,9 @@
+data "aws_region" "current" {}
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/2xx/balancer.tf]: unused data

File: modules/dashboard/modules/widgets/balancer/2xx/variables.tf [hunk 24]
```diff
@@ -0,0 +1,30 @@
+variable "balancer" {
+  type = string
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/2xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/2xx/variables.tf [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/2xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/2xx/variables.tf [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/2xx/variables.tf:16]: unused variable

File: modules/dashboard/modules/widgets/balancer/4xx/balancer.tf [hunk 27]
```diff
@@ -0,0 +1,9 @@
+data "aws_region" "current" {}
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/4xx/balancer.tf]: unused data

File: modules/dashboard/modules/widgets/balancer/4xx/variables.tf [hunk 28]
```diff
@@ -0,0 +1,30 @@
+variable "balancer" {
+  type = string
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/4xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/4xx/variables.tf [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/4xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/4xx/variables.tf [hunk 30]
```diff
@@ -0,0 +1,30 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/4xx/variables.tf:16]: unused variable

File: modules/dashboard/modules/widgets/balancer/5xx/balancer.tf [hunk 31]
```diff
@@ -0,0 +1,9 @@
+data "aws_region" "current" {}
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/5xx/balancer.tf]: unused data

File: modules/dashboard/modules/widgets/balancer/5xx/variables.tf [hunk 32]
```diff
@@ -0,0 +1,30 @@
+variable "balancer" {
+  type = string
+}
+
+# region
+variable "region" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/5xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/5xx/variables.tf [hunk 33]
```diff
@@ -0,0 +1,30 @@
...
+  type = object({
+    x : number
+    y : number
+    width : number
+    height : number
+  })
+}
+
+# stats
+variable "stat" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/5xx/variables.tf]: unused variable

File: modules/dashboard/modules/widgets/balancer/5xx/variables.tf [hunk 34]
```diff
@@ -0,0 +1,30 @@
...
+  type    = string
+  default = "Average"
+}
+
+variable "period" {
```
Comment (mrdntgrn) [inline modules/dashboard/modules/widgets/balancer/5xx/variables.tf:16]: unused variable

File: modules/dashboard/tests/widgets-balancer/1-example.tf (hunk 22 above)
Comment (aramkarapetian) [reply modules/dashboard/tests/widgets-balancer/1-example.tf:22]: This one is on purpose, cannot yet figure out easy way to create dependant balancer.

**Review by mrdntgrn — APPROVED** [review]

**Pull request merged**